server/src/main/java/org/opensearch/index/codec/composite/composite99/Composite99DocValuesWriter.java

```diff
@@ -66,7 +80,7 @@ public Composite99DocValuesWriter(DocValuesConsumer delegate, SegmentWriteState
         compositeFieldSet = new HashSet<>();
         segmentFieldSet = new HashSet<>();
         // TODO : add integ test for this
```

Comment:
Can you please address todo and add integ tests for this ? 

- where segment has no fields 
- segment has no doc count field
- segment has doc count field
- missing fields  ( segment has less fields than composite fields )

You can mark as done if you have covered these cases

Reply:
1 and 4 are already covered as a part of StarTreeDocValuesFormat Tests. That's how we got to know that StarTreeMetaData should contain all the dimension and metric names, because segment can have missing fields.

2 and 3 are specific to recently introduced doc count metric. Let me check the feasibility for those in this PR.

Reply:
Also, in what cases segment won't have doc_count field, if its the default metric? Please correct me if my understanding is wrong.

Reply:
In general segment won't have doc_count field , in which case we fill identity metric. In other cases, user will provide a value for pre-aggregated docs.